src/clear.rs

```diff
@@ -1,14 +1,27 @@
 use std::{collections, hash, ops::DerefMut, sync};
 
+/// Trait implemented by types which can be cleared in place, retaining any
+/// allocated memory.
+///
+/// This is essentially a generalization of methods on standard library
+/// collection types, including as [`Vec::clear`], [`String::clear`], and
+/// [`HashMap::clear`]. These methods drop all data stored in the collection,
+/// but retain the collection's heap allocation for future use. Types such as `BTreeMap` whose
+/// `clear` methods drops allocations should not implement this trait.
+///
```

Comment:
nit: this line is way longer than every other line in the comment:
```suggestion
/// but retain the collection's heap allocation for future use. Types such as
/// `BTreeMap`, whose `clear` methods drops allocations, should not 
/// implement this trait.
///
```